lib/helper.c

```diff
@@ -47,3 +47,18 @@ const char *byte2bitstr(uint8_t b)
 
     return str;
 }
+
+bool is_empty(uint8_t *buf, size_t size)
+{
+    if (size < sizeof(size_t)) {
+        for (size_t i = 0; i < size; i++) {
+            if (buf[i] != 0) {
+                return 0;
```

Comment:
Even though it's the same for C, I think a function of type `bool` should return `true` or `false` instead of `1` or `0`. The `0` could be confused with success, as used in many Zephyr APIs returning an `int` error code.